### ISSUE TITLE
Extend Operation Functionality

### ISSUE THREAD

**Issue by 0xflotus** [FlorianWendelborn/bitwise#44]
Can you please implement bit shift operations like _logical shift_, _arithmetic shift_ and _cyclic shift_ ?

- [x] Logical Shift
- [x] Arithmetic Shift
- [x] Cyclic Shift

**Comment by FlorianWendelborn** [comment]
Sorry, don’t have time for this right now, if you (or anyone else) want(s) to pick this up feel free to send a PR.

**Comment by FlorianWendelborn** [comment]
@0xflotus thanks for the PRs, I’m willing to get them merged soon. However, I left a couple questions on the individual pull requests

In general, this looks very mergeable already, pretty much only a few minor nitpicks open regarding public API.

**Comment by 0xflotus** [comment]
Are all commits fine?

**Comment by FlorianWendelborn** [comment]
@0xflotus yes, I think so, I’m unfortunately just very busy right now. It’s still on my radar though 😞
